inspirehep/modules/workflows/tasks/upload.py

```diff
@@ -62,11 +61,10 @@ def _get_updated_record(obj):
         # TODO: remove the skip files once labs becomes master
         record = InspireRecord.create(obj.data, id_=None, skip_files=True)
         # Create persistent identifier.
-        created_pid = inspire_recid_minter(str(record.id), record).pid_value
         # Now that we have a recid, we can properly download the documents
         record.download_documents_and_figures(src_records=[obj])
 
-        obj.data['control_number'] = created_pid
+        obj.data['control_number'] = record['control_number']
```

Comment:
could this function be rewritten in terms of `InspireRecord.create_or_update`?

Reply:
hmm this is for the workflow data, IMHO it's better to keep it separated